README.md

```diff
@@ -1,3 +1,156 @@
 # Spec Math
+## Overview
+Spec Math is a platform for performing operations on OpenAPI specifications. 
+[OpenAPI](https://github.com/OAI/OpenAPI-Specification) is the industry-standard
+format for machine-readable REST API descriptions.
 
-A tool that merges OpenAPI Specifications.
+## About this document
+This document contains an overview of this repository as well as a description of Spec Math.
+
+A goal of this document is to highlight the capabilities of Spec Math and
+describe its operations and functions so that you can determine if it might be useful to your project. 
+
+While this repository already contains a [Java implementation](library) of Spec Math, we acknowledge that
+your project might have different language needs and hope that this document (along with the
+[Java implementation](library)) will also provide
+enough information such that you could implement Spec Math in any language
+of your choosing. For more specific documentation and installation instructions, please look 
+at the respective folder in the directory structure
+described below.
+
+## Directory structure
+- The [library](library) folder contains the "Spec Math Library", an implementation of Spec Math in Java.
+- The [web](web) folder showcases an example usage of the Spec Math Library through an API which is connected
+ to the frontend application in the
+[app](app) folder.
+
+## Spec Math and the Spec Math Library
+
+### Background
+Specifications play a critical role in the API lifecycle. They can enforce a contract between client
+and server teams, easily generate documentation and client libraries, reduce the work of
+creating a proxy for an existing service and much more. The
+Spec Math project aims to afford even more power to API developers who use the OpenAPI
+specification. The [Spec Math Library](library) will enable them to do the following:
+
+- Merge two (or more in a future update) OpenAPI specifications into one.
+- Filter an OpenAPI specification into a new spec based on specified criteria.
+- Overlay an OpenAPI specification.
```

Comment:
"Overlay" seems unclear here. Perhaps add "by adding partial specs with domain-specific annotations"

Reply:
sounds good, I like that description! I think the word "superimposing" rather than "adding" might give some additional detail.